src/content/get-started/async.md

```diff
@@ -0,0 +1,374 @@
+    with the existing `dart:io` import:
+
+    ```dart
+    import 'dart:io';
+    import 'package:http/http.dart' as http; // Add this line
+    ```
+
+    This line imports the `http` package and gives it the alias `http`.
+    This allows you to refer to classes and functions within the `http`
+    package using `http.` (e.g., `http.Client`, `http.get`). The `as http`
+    part is a standard convention to avoid naming conflicts if another
+    imported library also has a similarly named class or function.
+
+### Task 3: Implement the `getWikipediaArticle` function
+
+You will now create a new function called `getWikipediaArticle` that
+handles fetching data from an external API. This function will be `async`
+because network requests are asynchronous operations.
+
+1.  **Define the function signature:**
+    Below your `main` function (and `printUsage` function), add the
+    following function signature.
+    The `Future<String>` return type indicates that this function will
+    eventually produce a `String` result, but not immediately, because
+    it's an asynchronous operation. The `async` keyword marks the
+    function as asynchronous, allowing you to use `await` inside it.
+
+    ```dart
+    // ... (your existing printUsage() function)
+
+    Future<String> getWikipediaArticle(String articleTitle) async {
+      // We'll add more code here soon
+    }
+    ```
+
+2.  **Construct the API URL and `http.Client`:**
+    Inside your new `getWikipediaArticle` function, create an `http.Client()`
+    instance and a `Uri` object. The `Uri` represents the endpoint of the
+    Wikipedia API you'll be calling to get an article summary.
+
+    Add these lines inside the `getWikipediaArticle` function:
+
+    ```dart
+    Future<String> getWikipediaArticle(String articleTitle) async {
+      final http.Client client = http.Client(); // Create an HTTP client
+      final Uri url = Uri.https(
+        'en.wikipedia.org', // Wikipedia API domain
+        '/api/rest_v1/page/summary/$articleTitle', // API path for article summary
+      );
+      // ...
+    }
+    ```
+
+    :::note
+    For applications making multiple HTTP requests or needing advanced
+    features like persistent connections, it's generally more efficient and
+    recommended to use an `http.Client()` instance and close it when no
+    longer needed.
+    :::
+
+3.  **Make the HTTP Request and handle the response:**
+    Now, use the `http` client to make an HTTP `GET` request to the URL
+    you just constructed. The `await` keyword pauses the execution of
+    `getWikipediaArticle` until the `client.get(url)` call completes and
+    returns an `http.Response` object.
+
+    After the request completes, check the `response.statusCode` to
+    ensure the request was successful (a status code of `200` means OK).
+    If successful, return the `response.body`, which contains the
+    fetched data (in this case, raw JSON). If the request failed, return
```

Comment:
```suggestion
    fetched data (in this case, raw JSON). If the request fails, return
```

Reply:
Same. I'm going to stop leaving comments. When you have a moment, make sure to look at all of these outdated comments and update the docs.